src/ir_York.cpp

```diff
@@ -0,0 +1,340 @@
+using irutils::minsToString;
+
+
+// Constants
+const uint16_t kYorkHdrMark = 4887;
+const uint16_t kYorkBitMark = 612;
+const uint16_t kYorkHdrSpace = 2267;
+const uint16_t kYorkOneSpace = 1778;
+const uint16_t kYorkZeroSpace = 579;
+const uint16_t kYorkFreq = 38000;  // Hz. (Guessing the most common frequency.)
+const uint16_t kYorkOverhead = 3;
+
+#if SEND_YORK
+/// Send a 17 Byte / 136 bit York A/C message.
+/// Status: ALPHA / Untested. The Checksum logic is missing
+/// @param[in] data An array of bytes containing the IR command.
+/// @param[in] nbytes Nr. of bytes of data in the array. (>=kStateLength)
+/// @param[in] repeat Nr. of times the message is to be repeated.
+void IRsend::sendYork(const uint8_t data[], const uint16_t nbytes,
+                      const uint16_t repeat) {
+  if (nbytes < kYorkStateLength)
+    return;
+  sendGeneric(kYorkHdrMark, kYorkHdrSpace,
+              kYorkBitMark, kYorkOneSpace,
+              kYorkBitMark, kYorkZeroSpace,
+              kYorkBitMark, kDefaultMessageGap,
+              data, kYorkStateLength,  // Bytes
+              kYorkFreq, false, repeat, kDutyDefault);  // false == LSB
+}
+#endif  // SEND_YORK
+
+#if DECODE_YORK
+/// Decode the supplied  message.
+/// Status: ALPHA / Tested, some values still are not mapped to the internal
+/// state of AC
+/// @param[in,out] results Ptr to the data to decode & where to store the decode
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return A boolean. True if it can decode it, false if it can't.
+bool IRrecv::decodeYork(decode_results *results, uint16_t offset,
+                        const uint16_t nbits, const bool strict) {
+  if (results->rawlen < 2 * nbits + kYorkOverhead - offset)
+    return false;  // Too short a message to match.
+  if (strict && nbits != kYorkBits)
+    return false;
+
+  uint16_t used = 0;
+
+  used = matchGeneric(results->rawbuf + offset, results->state,
+                      results->rawlen - offset, nbits,
+                      kYorkHdrMark, kYorkHdrSpace,
+                      kYorkBitMark, kYorkOneSpace,
+                      kYorkBitMark, kYorkZeroSpace,
+                      kYorkBitMark, kDefaultMessageGap,
+                      false, _tolerance, kMarkExcess,
+                      false);  // LSB
+  if (used == 0) return false;  // We failed to find any data.
+
+  // Succes
+  results->decode_type = decode_type_t::YORK;
+  results->bits = nbits;
+
+  return true;
+}
+#endif  // DECODE_YORK
+
+//
+//
+/// Class constructor
+/// @param[in] pin GPIO to be used when sending.
+/// @param[in] inverted Is the output signal to be inverted?
+/// @param[in] use_modulation Is frequency modulation to be used?
+IRYorkAc::IRYorkAc(const uint16_t pin, const bool inverted,
```

Comment:
I think you're missing a `IRYorkAc::stateReset()` function.
i.e. Call `setRaw(kYorkKnownGoodState);`